test/BUILD.bazel

```diff
@@ -1,23 +1,25 @@
-load("@rules_cc//cc:defs.bzl", "cc_test")
-load("@magic_enum//bazel:copts.bzl", "COPTS")
-
-# bazel central registry has a catch2 module, but is newer than the one included
-# in this repository
-cc_library(
-    name = "catch2",
-    includes = ["3rdparty/Catch2/include"],
-    hdrs = ["3rdparty/Catch2/include/catch2/catch.hpp"],
-    copts = COPTS,
-)
-
 _TESTS = [
     "test",
     "test_flags",
 ]
 
+_MSVC_FLAGS = ["/std:c++17", "/permissive-"]
+_COPTS = select({
+    "//conditions:default": ["-std=c++17"],
+    "@bazel_tools//tools/cpp:clang-cl": _MSVC_FLAGS,
+    "@bazel_tools//tools/cpp:msvc": _MSVC_FLAGS, 

```

Comment:
These options don't work unless specifying platforms as far as I know. Example from my previous bzlmod PR https://github.com/Neargye/magic_enum/pull/254/files#diff-e13b51fb7feeb38d3fb7dee88724107c57fb3d4805068b5f8d57a2d0900fb0dc. It'll just fallback to the default.

e.g.
```
cl : Command line warning D9002 : ignoring unknown option '-std=c++17'
```
Followed by many errors.

Reply:
Ah shoot. Sorry I missed that. The other one worked without platforms, I take it? Presumably the toolchains set the @bazel_tools//tools/cpp:compiler flag internally?

I think the right move, then, is to re-add the bazelmod rules_cc dependency for tests (only). Seems cleaner than duplicating the config settings and still accomplishes all the previous goals around consuming the library itself. I'll push up a commit that does just that.